File: Product_Management_Backend/src/main/java/com/Daniel/Product_Management_Backend/service/ProductService.java
package com.Daniel.Product_Management_Backend.service;

import com.Daniel.Product_Management_Backend.model.Product;
import org.springframework.context.annotation.Bean;

import java.util.List;

public interface ProductService {

    public Product saveProduct(Product product);

    public List<Product> getAllProduct();

    public Product getProductById(Integer id);

    public String deleteProduct(Integer id);

    public Product editProduct(Product product,Integer id);

}

File: Product_Management_Frontend/product_ui/src/service/productService.js
const API_URL = "http://lochalhost:8080";

class ProductService{

    saveProduct(product){
        return axios.post(API_URL+"/saveProduct", product);
    }

    getAllProduct(product){
        return axios.get(API_URL+"/");
    }

    getProductById(id){
        return axios.get(API_URL+"/", id);
    }

    deleteProduct(id){
        return axios.get(API_URL+"/deleteProduct/", id);
    }
    
    editProduct(product){
        return axios.post(API_URL+"/"+product.id, product);
    }



}

export default new ProductService